packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState } from "react"
+import { render, screen, fireEvent, waitFor } from "@testing-library/react"
+import "@testing-library/jest-dom"
+import userEvent from "@testing-library/user-event"
+import { RichTextEditor, EditorContentArray } from "./"
+
+// This helper is needed to simulate selection of a component since we
+// cannot userEvent.type with contenteditable
+const getSelectionOfNode = node => {
+  // Clear any current selection
+  const selection = window.getSelection()
+  selection?.removeAllRanges()
+
+  // Select paragraph
+  const range = document.createRange()
+  range.selectNodeContents(node)
+  selection?.addRange(range)
+}
+
+const TestRTE = args => {
+  const { rteMockData, ...rest } = args
+  const [rteData, setRTEData] = useState<EditorContentArray>(
+    args.rteMockData || []
+  )
+  return (
+    <>
+      <RichTextEditor
+        value={rteData}
+        onChange={data => setRTEData(data)}
+        {...rest}
+      />
+    </>
+  )
+}
+
+describe("RTE receives list controls", () => {
+  const defaultListArgs = {
+    labelText: "List RTE",
+    rows: 3,
+    controls: [
+      { name: "orderedList", group: "list" },
+      { name: "bulletList", group: "list" },
+    ],
+  }
+
+  it("renders list buttons when receiving a list controls", () => {
+    render(<TestRTE {...defaultListArgs} />)
+
+    const bulletButton = screen.getByRole("button", { name: "Bullet List" })
+    const orderedButton = screen.getByRole("button", { name: "Numbered List" })
+
+    expect(bulletButton && orderedButton).toBeTruthy()
+  })
+
+  it("renders indent buttons when receiving a list controls", () => {
+    render(<TestRTE {...defaultListArgs} />)
+
+    const decreaseIndentBtn = screen.getByRole("button", {
+      name: "Decrease indent",
+    })
+    const increaseIndentBtn = screen.getByRole("button", {
+      name: "Increase indent",
+    })
+
+    expect(decreaseIndentBtn && increaseIndentBtn).toBeTruthy()
+  })
+
+  describe("Creating list nodes with buttons", () => {
+    it("will create a <ul> when user clicks the bullet list button", async () => {
+      render(<TestRTE {...defaultListArgs} />)
+
+      // We would use userEvent.type but contenteditable is not supported
+      // see thread: https://github.com/testing-library/user-event/issues/230
+      fireEvent.focus(screen.getByRole("textbox", { name: "List RTE" }), {
+        target: { textContent: "this will be a ul" },
+      })
+
+      await waitFor(() => {
+        userEvent.click(screen.getByRole("button", { name: "Bullet List" }))
+        screen.getByRole("list")
+
+        expect(screen.getByRole("list")).toBeInTheDocument()
+        expect(document.querySelectorAll("ul").length).toBeGreaterThan(0)
```

Comment:
Is the second assertion here redundant? Wouldn't the first one fail if there's no ul?

Reply:
That is true, hang up from when I was tryign to be thorough to verify it was working, the less assertions the better hey